File: halcyon-web/src/main/resources/static/js/controller/oauth2/oauthAccessTokenController.js
//控制层
app.controller("oauthAccessTokenController", function ($scope, $controller, oauthAccessTokenService) {
    //继承
    $controller("baseController", {$scope: $scope});

    //初始化
    $scope.initForAdd = function () {
        $scope.oauthAccessToken = {};
        $scope.getEmpOptions();
    }
    $scope.initForUpdate = function (id) {
        oauthAccessTokenService.queryById(id).success(function (data) {
            $scope.oauthAccessToken = data;
        });
        $scope.getEmpOptions();
    }

    //crud
    //查询,初始化查询条件为空
    $scope.condition = {};
    $scope.query = function (pageNum, pageSize) {
        oauthAccessTokenService.query(pageNum, pageSize, $scope.condition).success(function (responseData) {
            $scope.parseResponse(responseData);
        });
    }
    //保存
    $scope.save = function () {
        oauthAccessTokenService.save($scope.oauthAccessToken).success(function (responseData) {
            $scope.parseResponse(responseData);
        });
    }
    //删除
    $scope.delete = function () {
        $scope.deleteConfirm(deleteRows);
    }

    function deleteRows() {
        oauthAccessTokenService.delete($scope.selectedIds).success(function (responseData) {
            $scope.parseResponse(responseData);
        });
    }

    //获取下拉框数据
    //防止ajax异步导致为空，声明一个空的
    $scope.oauthAccessTokenOptions = {data: []};
    $scope.getOptions = function () {
        oauthAccessTokenService.getOptions().success(function (data) {
            $scope.oauthAccessTokenOptions = {data: data};
        });
    }

});

File: halcyon-oauth2/src/main/java/com/marshal/halcyon/oauth2/controller/OauthAccessTokenController.java
package com.marshal.halcyon.oauth2.controller;

import com.marshal.halcyon.core.entity.ResponseData;
import com.marshal.halcyon.core.controller.BaseController;
import com.marshal.halcyon.core.util.ResponseUtil;
import com.marshal.halcyon.oauth2.entity.OauthAccessToken;
import com.marshal.halcyon.oauth2.service.OauthAccessTokenService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

/**
 * @auth: Marshal
 * @date: 2019/3/6
 * @desc:
 */
@RestController
@RequestMapping("/oauth2/accessToken")
public class OauthAccessTokenController extends BaseController {

    @Autowired
    OauthAccessTokenService oauthAccessTokenService;

    @RequestMapping("/query")
    public ResponseData query(@RequestBody OauthAccessToken condition,
                              @RequestParam int pageNum,
                              @RequestParam int pageSize) {
        return new ResponseData(oauthAccessTokenService.select(condition, pageNum, pageSize));
    }

    @RequestMapping("/submit")
    public ResponseData save(@RequestBody OauthAccessToken oauthClientDetails) {
        if (!getValidator().hasError(oauthClientDetails)) {
            return new ResponseData(false, getValidator().getErrors(oauthClientDetails));
        }
        oauthAccessTokenService.save(oauthClientDetails);
        return ResponseUtil.responseOk("保存成功");
    }

    @RequestMapping("/remove")
    public ResponseData delete(@RequestParam("selectedIds") Long[] selectedIds) {
        oauthAccessTokenService.batchDelete(selectedIds);
        return ResponseUtil.responseOk("删除成功");
    }

    @RequestMapping("/queryById")
    public OauthAccessToken queryById(@RequestParam Long id) {
        return oauthAccessTokenService.selectByPrimaryKey(id);
    }
}
